### PR TITLE
Jvm remote

### PR DESCRIPTION
**Pull request by astj**
mackerel-plugin-jvm has `-host` and `-port` option but they don't work as expected.
I'd like change those options to `-remote` option and make it work with rmi protocol.

In target host
```
/tmp$ cat jstatd.policy
grant codebase "file:${java.home}/../lib/tools.jar" {
  permission java.security.AllPermission;
};
/tmp$ jstatd -J-Djava.security.policy=/tmp/jstatd.policy -p [Port] -J-Djava.rmi.server.hostname=[Hostname]
```

In monitoring host:

```
$ ./mackerel-plugin-jvm -javaname XXX -remote [targetHostName:targetPort]
```

### CODE REVIEW

File: mackerel-plugin-jvm/lib/jvm.go [hunk 1]
```diff
@@ -290,10 +304,45 @@ func (m JVMPlugin) GraphDefinition() map[string]mp.Graphs {
...
+	return lvmid
+}
+
+func generateRemote(remote, host string, port int) string {
+	if remote == "" {
+		if host == "" {
+			if port != 0 {
+				// for backward compatibility
+				hostPort := fmt.Sprintf("localhost:%d", port)
+				return hostPort
```
Comment (itchyny) [inline mackerel-plugin-jvm/lib/jvm.go]: [nit] return fmt.Sprintf...

File: mackerel-plugin-jvm/lib/jvm.go [hunk 2]
```diff
@@ -290,10 +304,45 @@ func (m JVMPlugin) GraphDefinition() map[string]mp.Graphs {
...
+	}
+	return remote
+}
+
 // Do the plugin
 func Do() {
-	optHost := flag.String("host", "localhost", "Hostname")
-	optPort := flag.String("port", "1099", "Port")
+	optHost := flag.String("host", "", "jps/jstat target hostname [deprecated]")
+	optPort := flag.Int("port", 0, "jps/jstat target port [deprecarted]")
```
Comment (itchyny) [inline mackerel-plugin-jvm/lib/jvm.go]: [typo] deprecated, not deprecarted

**Comment by itchyny** [comment]
LGTM

**Comment by astj** [comment]
Thanks!

**Pull request merged**